definition/definition.go

```diff
@@ -31,14 +31,43 @@ type Chain interface {
 // carry on, call Chain.Continue() and pass the context.
 type Middleware func(context.Context, Chain) error
 
+// Operator is used to operate an object and return an object as replacement.
+//
+// For example:
+//  A converter:
+//    type ConverterForAnObject struct{}
+//    func (c *ConverterForAnObject) Kind() {return "converter"}
+//    func (c *ConverterForAnObject) In() reflect.Type {return definition.TypeOf(&ObjectV1{})}
+//    func (c *ConverterForAnObject) Out() reflect.Type {return definition.TypeOf(&ObjectV2{})}
+//    func (c *ConverterForAnObject) Operate(ctx context.Context, object interface{}) (interface{}, error) {
+//        objV2, err := convertObjectV1ToObjectV2(object.(*ObjectV1))
+//        return objV2, err
+//    }
+//
+//  A validator:
+//    type ValidatorForAnObject struct{}
+//    func (c *ValidatorForAnObject) Kind() {return "validator"}
+//    func (c *ValidatorForAnObject) In() reflect.Type {return definition.TypeOf(&Object{})}
+//    func (c *ValidatorForAnObject) Out() reflect.Type {return definition.TypeOf(&Object{})}
+//    func (c *ValidatorForAnObject) Operate(ctx context.Context, object interface{}) (interface{}, error) {
+//        if err := validate(object.(*Object)); err != nil {
+//            return nil, err
+//        }
+//        return object, nil
+//    }
 type Operator interface {
```

Comment:
If I have multiple operators, what is the calling order for operator?  Another question, for incoming request, are all param operators be called after all middleware?  can you elaborate a little bit or point me to the code,

Reply:
Operators execute one by one.
The workflow could explain it:
```
// Workflow:
//            Service.ServeHTTP()
//          ----------------------
//          ↓                    ↑
// |-----Filters------|          ↑
//          ↓                    ↑
// |---Router Match---|          ↑
//          ↓                    ↑
// |-------------Middlewares------------|
//          ↓                    ↑
// |-------------Executor---------------|
//          ↓                    ↑
// |-ParameterGenerators-|-DestinationHandlers-|
//          ↓                    ↑
// |------------User Function-----------|
```
Parameter operators are called by ParameterGenerators.
Destination operators are called by DestinationHandlers.